### PR TITLE
Add page to create channel

### LINKED ISSUE #16: As a logged in user I would like to be able to create a channel
The most minimal form to create a channel.  

- [x] Implement backend API (PR #39)
- [ ] Implement frontend UI
- [ ] Integrate frontend UI with API

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #16 

#### What's this PR do?
Adds a page to create a channel

#### How should this be manually tested?
Go to http://localhost:8063/manage/channel/new and fill in the form. Verify you're redirected to your new channel.

#### Any background context you want to provide?
No validation yet

### CODE REVIEW

File: static/js/containers/admin/AdminPage.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  props: {
+    match: Match,
+  }
+
+  render() {
+    const { match } = this.props;
+    return (
+      <div className="admin double-column">
+        <Card>
+          <h1>Admin</h1>
```
Comment (aliceriot) [inline static/js/containers/admin/AdminPage.js]: should we maybe add a `title` prop to the `Card` component? I'm thinking we may want to style that in a consistent way across the app (just occurred to me, not necessary something to do in this PR)

Comment (rhysyngsun) [inline static/js/containers/admin/AdminPage.js]: Yeah, I think something like that would be good eventually.

**Comment by noisecapella** [comment]
After I successfully create a channel I'm redirected to the channel page but I get an error page:

screenshot from 2017-07-31 14-31-08

File: static/js/components/admin/ChannelEditForm.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+// @flow
+import React from 'react';
+
+export default class ChannelEditForm extends React.Component {
+  props: {
+    onSubmit: Function,
+    onUpdate: Function,
+    channel:  ?Object,
```
Comment (noisecapella) [inline static/js/components/admin/ChannelEditForm.js]: Can you use the `Channel` type here?

File: static/js/components/admin/ChannelEditForm_test.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        assert.isNotOk(onSubmit.called);
+        wrapper.find('form').simulate('submit');
+        assert.isOk(onSubmit.called);
+        assert.isNotOk(onUpdate.called);
+      });
+    });
+
+    describe('onUpdate', () => {
+      for (let inputName of ['title', 'name', 'public_description']) {
+        it(`should be called when ${inputName} input is modifed`, () => {
```
Comment (noisecapella) [inline static/js/components/admin/ChannelEditForm_test.js]: Should be `modified`

File: static/js/containers/admin/AdminPage.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+          <Route path={`${match.url}/channel/new`} component={CreateChannelPage}/>
```
Comment (noisecapella) [inline static/js/containers/admin/AdminPage.js]: Should we use the functions in `lib/url.js` instead for this kind of thing?

File: static/js/containers/admin/CreateChannelPage.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }));
+  }
+
+  onSubmit = (e: Object) => {
+    const { dispatch, history, channelForm } = this.props;
+
+    e.preventDefault();
+
+    dispatch(actions.channels.post(channelForm.value)).then((channel) => {
+      history.push(`/channel/${channel.name}`);
```
Comment (noisecapella) [inline static/js/containers/admin/CreateChannelPage.js]: Can we make a function in `lib/url.js` to create this URL?

File: static/js/lib/api.js [hunk 6]
```diff
@@ -17,6 +18,15 @@ export function getChannel(channelName: string): Promise<Channel> {
   return fetchJSONWithCSRF(`/api/v0/channels/${channelName}/`);
 }
 
+export function createChannel(channel: Channel): Promise<Channel> {
+  return fetchJSONWithCSRF(`/api/v0/channels/`, {
+    method: POST,
+    body: JSON.stringify({
+      ...channel
```
Comment (noisecapella) [inline static/js/lib/api.js]: Why are we using the spread syntax here?

File: static/js/lib/channels_test.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+import { newChannel } from './channels';
+
+describe('Channel utils', () => {
+  it('should return a new channel with empty values and public type', () => {
+    assert.deepEqual(newChannel(), {
+      'title': '',
+      'name': '',
+      'public_description': '',
+      'channel_type': 'public',
```
Comment (noisecapella) [inline static/js/lib/channels_test.js]: Can we make constants for `public` and `private`?

File: static/js/lib/form_actions.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+export const deriveReducer = (name: string, newDefaults: () => Object): FormReducer => {
+  const actionTypes = deriveActionTypes(name);
+  return (state: FormValue = INITIAL_STATE, action: Action<any, null>): FormValue => {
+    switch(action.type) {
+    case actionTypes.reset:
+    case actionTypes.create:
+      return {
+        ...state,
```
Comment (noisecapella) [inline static/js/lib/form_actions.js]: What about `errors`? It seems like we should clear it here

File: static/js/lib/form_actions_test.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+      sandbox = sinon.sandbox.create();
+      store = configureTestStore(combineReducers({ channels }));
+      dispatchThen = store.createDispatchThen(state => state.channels);
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    //
```
Comment (noisecapella) [inline static/js/lib/form_actions_test.js]: Can you remove this line?

File: static/js/lib/form_actions.js (hunk 8 above)
Comment (rhysyngsun) [inline static/js/lib/form_actions.js]: `errors` is covered by the spread, should I get rid of the spread and just be explicit about it?

Comment (noisecapella) [inline static/js/lib/form_actions.js]: It's not the explicitness, I'm wondering about if we should replace `errors` here with `{}` to clear the errors

File: static/js/containers/admin/AdminPage.js (hunk 4 above)
Comment (rhysyngsun) [inline static/js/containers/admin/AdminPage.js]: Not in this case because it's the route config, not a link to a page.

File: static/js/lib/api.js (hunk 6 above)
Comment (rhysyngsun) [inline static/js/lib/api.js]: Yeah, that's less than useful, it also isn't guarding against the other channel properties which could cause a 4xx.

File: static/js/lib/form_actions.js (hunk 8 above)
Comment (rhysyngsun) [inline static/js/lib/form_actions.js]: Good point.

Comment (rhysyngsun) [inline static/js/lib/form_actions.js]: I won't have a good way of testing that errors gets reset until I add support for that, which will be in a separate PR, but worth fixing this issue now.

**Comment by rhysyngsun** [comment]
@noisecapella the diff is probably hosed now since I rebased. Look at this commit to see what I fixed relative your feedback: https://github.com/mitodl/open-discussions/pull/73/commits/ce770eccb011bf87b3e1785b24cef3d08e837155

**Comment by rhysyngsun** [comment]
@noisecapella didn't see the error until just now. This would've been caused by an error from the backend so you might have to check your network tab or backend logs for more details

**Comment by noisecapella** [comment]
I just checked out the newest commit, it looks like the error is gone now
